app/src/main/java/social/entourage/android/DrawerActivity.java

```diff
@@ -993,27 +993,29 @@ private void addPushNotification(Message message) {
     // ----------------------------------
 
     public void showEditActionZoneFragment() {
-        showEditActionZoneFragment(false, null);
+        showEditActionZoneFragment( null);
     }
 
-    public void showEditActionZoneFragment(boolean forced, UserEditActionZoneFragment.FragmentListener extraFragmentListener) {
-        if (editActionZoneShown && !forced) return;
+    public void showEditActionZoneFragment(UserEditActionZoneFragment.FragmentListener extraFragmentListener) {
         AuthenticationController authenticationController = EntourageApplication.get().getEntourageComponent().getAuthenticationController();
-        if (authenticationController.isAuthenticated()) {
+        if (!authenticationController.isAuthenticated()) return;
             User me = authenticationController.getUser();
+        if(me==null) return;
+
             UserPreferences userPreferences = authenticationController.getUserPreferences();
-            if (me != null && userPreferences != null) {
-                boolean noAddress = me.getAddress() == null || me.getAddress().getDisplayAddress() == null || me.getAddress().getDisplayAddress().length() == 0;
-                if ((forced || !userPreferences.isEditActionZoneShown()) && noAddress) {
-                    UserEditActionZoneFragment userEditActionZoneFragment = UserEditActionZoneFragment.newInstance(null);
-                    userEditActionZoneFragment.addFragmentListener(this);
-                    userEditActionZoneFragment.addFragmentListener(extraFragmentListener);
-                    userEditActionZoneFragment.setFromLogin(true);
-                    userEditActionZoneFragment.show(getSupportFragmentManager(), UserEditActionZoneFragment.TAG);
-                }
-            }
-        }
-        editActionZoneShown = true;
+        if(userPreferences== null) return;
+
+        if (me.isEditActionZoneShown()
```

Comment:
You should extract this complex condition into a variable to make it clearer to read